server/src/models/database/socobo-user.ts

```diff
@@ -1,16 +1,64 @@
+import { IsEmail, IsNotEmpty, MinLength, ValidateIf } from "class-validator";
 import { ProviderType } from "./../enums/provider-type";
 import { Role } from "./../enums/role";
+import { ValidationGroup } from "./../enums/validation-group";
 
 export class SocoboUser {
+
   public id: number;
+
+  @ValidateIf((o) => o.email === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @MinLength(5, {
+    groups: [ ValidationGroup.LOGIN ]
+  })
   public username: string;
+
+  @ValidateIf((o) => o.username === "", {
+    groups: [ ValidationGroup.LOGIN ]
+  })
+  @IsNotEmpty({
+    groups: [ ValidationGroup.REGISTRATION ]
+  })
+  @IsEmail({}, {
+    groups: [
+      ValidationGroup.LOGIN,
+      ValidationGroup.REGISTRATION
+    ]
+  })
   public email: string;
+
+  @IsNotEmpty({
+    groups: [
+      ValidationGroup.LOGIN,
+      ValidationGroup.REGISTRATION
+    ]
+  })
+  @MinLength(8, {
+    groups: [
```

Comment:
If I have something like this:
{
	"username": "",
	"email": "",
	"password": "testtest"
}
I got 
email must be an email

In some way this is correct, but is it valid that the email is empty during the login if I provide a email property? I think if there is a email property in the body it should not be empty as with the username